### ISSUE TITLE
Video playback hangs for JW Player (QT 5 & 6)

### ISSUE THREAD

**Issue by Narice** [qutebrowser/qutebrowser#7513]
**Version info**:
qutebrowser v2.5.2
Git commit: 
Backend: QtWebEngine 5.15.11, based on Chromium 87.0.4280.144
Qt: 5.15.7

**Does the bug happen if you start with `--temp-basedir`?**:
yes

**Description**
In under a minute, a video with sound activated and using the jw-player will hang.
This isn't happening with other video providers such as the youtube player.

For some context, here are all parts of my setup that can be relevent:
Wayland, Hyprland, Pipewire (providing pulseaudio server).

This doesn't happen in the Gnome Web browser, nor in Firefox.

**How to reproduce**

- Go to https://www.jwplayer.com/
- A demo video will start
- Activate the sound
- wait until 30seconds in
- The video should show the following:
![image](https://user-images.githubusercontent.com/31109099/206516126-d5e74ca2-a2ff-4069-b733-1b5633c4ff0e.png)
it is very consistant, if it doesn't happen the first time, then do it a second time. My reasoning is that is it fairly consistent from the moment you unmute, and as the video is short, if you unmute it too late, it won't crash.

Here is the 244002 error reason from [jw player](https://docs.jwplayer.com/players/docs/jw8-player-errors-reference#media-playback-shaka): The video element reported an error during DASH playback.

**Comment by The-Compiler** [comment]
Works fine for me, tried 3 times (though on X11 and with Pulseaudio).

Does the `:devtools` console have anything to say about it? Any output when you launch in a terminal with `--temp-basedir`? Any chance you could try in Chromium, and/or another QtWebEngine browser, such as KDE Falkon?

**Comment by Narice** [comment]
Just tried with Falkon and the issue is present there too, which would mean that it is an upstream issue?
Here is qutebrowser's output:
```
18:56:21 WARNING: Resource type ResourceTypeWebSocket not found in RequestInterceptor dict.
terminate called after throwing an instance of 'std::logic_error'
  what():  basic_string::_M_construct null not valid
Received signal 6
#0 0x7f36243a2269 base::debug::CollectStackTrace()
#1 0x7f36242f3cb6 base::debug::StackTrace::StackTrace()
#2 0x7f36243a1c53 base::debug::(anonymous namespace)::StackDumpSignalHandler()
#3 0x7f361d83dbf0 (/nix/store/4nlgxhb09sdr51nc9hdm8az5b08vzkgx-glibc-2.35-163/lib/libc.so.6+0x3dbef)
#4 0x7f361d88abc7 __pthread_kill_implementation
#5 0x7f361d83db46 __GI_raise
#6 0x7f361d8284b5 __GI_abort
#7 0x7f361dca996a _ZN9__gnu_cxx27__verbose_terminate_handlerEv.cold
#8 0x7f361dcb4f9a __cxxabiv1::__terminate()
#9 0x7f361dcb5005 std::terminate()
#10 0x7f361dcb5258 __cxa_throw
#11 0x7f361dcac1d1 std::__throw_logic_error()
#12 0x7f362194c93c std::__cxx11::basic_string<>::_M_construct<>()
#13 0x7f362194cc57 alsa_util::CloseDevice()
#14 0x7f362194aa0f media::AlsaPcmOutputStream::Close()
#15 0x7f3621940062 media::AudioOutputDispatcherImpl::CloseIdleStreams()
#16 0x7f362439bbcc base::RetainingOneShotTimer::RunUserTask()
#17 0x7f362439b88e base::internal::Invoker<>::RunOnce()
#18 0x7f3624353df9 base::DefaultDelayedTaskHandleDelegate::RunTask()
#19 0x7f36243534bc base::internal::Invoker<>::RunOnce()
#20 0x7f3624355853 base::TaskAnnotator::RunTaskImpl()
#21 0x7f362436d624 base::sequence_manager::internal::ThreadControllerWithMessagePumpImpl::DoWorkImpl()
#22 0x7f362436ded9 base::sequence_manager::internal::ThreadControllerWithMessagePumpImpl::DoWork()
#23 0x7f3624312d39 base::MessagePumpDefault::Run()
#24 0x7f362436bdc6 base::sequence_manager::internal::ThreadControllerWithMessagePumpImpl::Run()
#25 0x7f362433ee5e base::RunLoop::Run()
#26 0x7f3623bcc022 content::UtilityMain()
#27 0x7f3623bc59d9 content::RunOtherNamedProcessTypeMain()
#28 0x7f3623bc5ba9 content::ContentMainRunnerImpl::Run()
#29 0x7f3623bc29fc content::RunContentProcess()
#30 0x7f3623bc3501 content::ContentMain()
#31 0x7f3620bbba45 QtWebEngineCore::processMain()
#32 0x000000402222 main
#33 0x7f361d82924e __libc_start_call_main
#34 0x7f361d829309 __libc_start_main_alias_2
#35 0x000000402385 _start
  r8: 00007f3615c9b5c8  r9: 00007ffdb6b84e70 r10: 0000000000000008 r11: 0000000000000246
 r12: 00007f35fc00b8e0 r13: 00007f361dcc9850 r14: 0000000000000006 r15: 00007ffdb6b85a90
  di: 000000000006e217  si: 000000000006e217  bp: 000000000170a058  bx: 000000000006e217
  dx: 0000000000000006  ax: 0000000000000000  cx: 00007f361d88abc7  sp: 00007ffdb6b85270
  ip: 00007f361d88abc7 efl: 0000000000000246 cgf: 002b000000000033 erf: 0000000000000000
 trp: 0000000000000000 msk: 0000000000000000 cr2: 0000000000000000
[end of stack trace]
```
And here is falkon's output:
```
[454176:4:1208/185827.132105:ERROR:batching_media_log.cc(38)] MediaEvent: {"error":"audio render error"}
[454176:1:1208/185827.133962:ERROR:batching_media_log.cc(35)] MediaEvent: {"pipeline_error":19}
```
Should this issue be closed and should an issue elsewhere be opened instead? (I have no clue where this issue is coming from tbh :sweat_smile:)

**Comment by The-Compiler** [comment]
No idea what that means exactly, but clearly not something qutebrowser can do anything about, yep. Thus closing this as an upstream issue in either [QtWebEngine](https://wiki.qt.io/QtWebEngine), or Chromium.

I try to report bugs upstream when I can easily reproduce them, but unfortunately this wasn't the case here. All I was able to find is an old Chromium bug report which was closed because of too little information: [1097550 - error code 256 - chromium](https://bugs.chromium.org/p/chromium/issues/detail?id=1097550&q=basic_string%3A%3A_M_construct%20null%20not%20valid&can=1)

I suggest you try with Chromium, and then either report this there, or report it to the [Qt Bugtracker](https://bugreports.qt.io/secure/Dashboard.jspa). For the latter, please use "Project: Qt (QTBUG)"  and "Component: WebEngine" there. After reporting the bug, it'd be good if you could post the link to it here!

**Comment by Narice** [comment]
Thanks for the help!
